pkg/model/config/config.go

```diff
@@ -46,22 +49,26 @@ type Config struct {
 
 	// Layout contains a key specifying which plugin created a project.
 	Layout string `json:"layout,omitempty"`
+
+	// ExtraFields is an arbitrary YAML blob that can be used by non-kubebuilder
+	// plugins for plugin-specific configure.
+	ExtraFields map[string]interface{} `json:"plugins,omitempty"`
```

Comment:
non-blocking: can we wrap this in a type so that people don't depend on the exact deserialization internals?  If we want people to use types and the DecodeExtraFields here, we might want to wrap `interface{}` in something.

Reply:
Sounds reasonable. I'll make a follow-up PR so we can discuss specifics there.